### PR TITLE
Bugfixing redirects

### PR DESCRIPTION
**Pull request by nupplaphil**
- Adding App->redirect if both redirects are possible

Fixing #6019 and parts of #5996

### CODE REVIEW

File: mod/randprof.php [hunk 1]
```diff
@@ -15,11 +15,7 @@ function randprof_init(App $a)
 	if ($x) {
 		$link = Contact::magicLink($x);
 		// @TODO making the return of magicLink save to use either externalRedirect or internalRedirect
```
Comment (MrPetovan) [inline mod/randprof.php]: This can be removed.

File: src/App.php [hunk 2]
```diff
@@ -2006,4 +2006,21 @@ public function internalRedirect($toUrl = '', $ssl = false)
...
+	/**
+	 * Redirects to another URL in case
+	 * Should only be used if it isn't clear if the URL is either internal or external
+	 *
+	 * @param string $toUrl The target URL
+	 *
+	 */
+	public function redirect($toUrl)
+	{
+		if (filter_var($toUrl, FILTER_VALIDATE_URL))
```
Comment (MrPetovan) [inline src/App.php]: Code standards: Please move the opening curly brace on the same line as the `if`.

File: src/App.php [hunk 3]
```diff
@@ -2006,4 +2006,20 @@ public function internalRedirect($toUrl = '', $ssl = false)
 		$redirectTo = $this->getBaseURL($ssl) . '/' . ltrim($toUrl, '/');
 		Core\System::externalRedirect($redirectTo);
 	}
+
+	/**
+	 * Redirects to another URL in case
```
Comment (MrPetovan) [inline src/App.php]: in case?

Comment (nupplaphil) [reply src/App.php]: Ah damn, bad translation of a german meaning ^^

**Pull request merged**